internal/exec/terraform_clean.go

```diff
@@ -0,0 +1,432 @@
+	absBasePath, err := filepath.Abs(basePath)
+	if err != nil {
+		fmt.Printf("Error getting absolute path for basePath: %v\n", err)
+		return "", err
+	}
+	absComponentPath, err := filepath.Abs(componentPath)
+	if err != nil {
+		fmt.Printf("Error getting absolute path for componentPath: %v\n", err)
+		return "", err
+	}
+
+	relPath, err := filepath.Rel(absBasePath, absComponentPath)
+	if err != nil {
+		fmt.Printf("Error getting relative path: %v\n", err)
+		return "", err
+	}
+
+	return filepath.Base(absBasePath) + "/" + relPath, nil
+}
+func confirmDeleteTerraformLocal(message string) (confirm bool, err error) {
+	confirm = false
+	t := huh.ThemeCharm()
+	cream := lipgloss.AdaptiveColor{Light: "#FFFDF5", Dark: "#FFFDF5"}
+	purple := lipgloss.AdaptiveColor{Light: "#5B00FF", Dark: "#5B00FF"}
+	t.Focused.FocusedButton = t.Focused.FocusedButton.Foreground(cream).Background(purple)
+	t.Focused.SelectSelector = t.Focused.SelectSelector.Foreground(purple)
+	t.Blurred.Title = t.Blurred.Title.Foreground(purple)
+	confirmPrompt := huh.NewConfirm().
+		Title(message).
+		Affirmative("Yes!").
+		Negative("No.").
+		Value(&confirm).WithTheme(t)
+	if err := confirmPrompt.Run(); err != nil {
+		if err == huh.ErrUserAborted {
+			return confirm, fmt.Errorf("Mission aborted")
+		}
+		return confirm, err
+	}
+
+	return confirm, nil
+}
+
+// DeletePathTerraform deletes the specified file or folder. with a checkmark or xmark
+func DeletePathTerraform(fullPath string, objectName string) error {
+	if _, err := os.Stat(fullPath); os.IsNotExist(err) {
+		xMark := lipgloss.NewStyle().Foreground(lipgloss.Color("9")).SetString("x")
+		fmt.Printf("%s Cannot delete %s: path does not exist", xMark, objectName)
+		fmt.Println()
+		return err
+	}
+	// Attempt to delete the file or folder If the path does not exist, RemoveAll returns nil (no error)
+	err := os.RemoveAll(fullPath)
+	if err != nil {
+		xMark := lipgloss.NewStyle().Foreground(lipgloss.Color("9")).SetString("x")
+		fmt.Printf("%s Error deleting %s", xMark, objectName)
+		fmt.Println()
+		return err
+	}
+
+	checkMark := lipgloss.NewStyle().Foreground(lipgloss.Color("42")).SetString("✓")
+	fmt.Printf("%s deleted %s", checkMark, objectName)
+	fmt.Println()
+	return nil
+}
+
+// confirmDeletion prompts the user for confirmation before deletion.
+func confirmDeletion(cliConfig schema.CliConfiguration) (bool, error) {
+	message := "Are you sure?"
+	confirm, err := confirmDeleteTerraformLocal(message)
+	if err != nil {
+		return false, err
+	}
+	if !confirm {
+		u.LogWarning(cliConfig, "Mission aborted.")
+		return false, nil
+	}
+	return true, nil
+}
+
+// deleteFolders handles the deletion of the specified folders and files.
+func deleteFolders(folders []Directory, relativePath string) {
+	for _, folder := range folders {
+		for _, file := range folder.Files {
+			path := filepath.ToSlash(filepath.Join(relativePath, file.Name))
+			if file.IsDir {
+				DeletePathTerraform(file.FullPath, path+"/")
```

Comment:
Is this safe or should we use OS specific folder delimiter?

Reply:
its just for log fullPath used for delete